fuzz/fuzz_targets/client_initial.rs

```diff
@@ -17,18 +17,23 @@ fuzz_target!(|data: &[u8]| {
     };
 
     let mut client = default_client();
-    let ci = client.process(None, now()).dgram().expect("a datagram");
+    let ci = client
+        .process_alloc(None, now())
+        .dgram()
+        .expect("a datagram");
     let Some((header, d_cid, s_cid, payload)) = decode_initial_header(&ci, Role::Client) else {
         return;
     };
     let (aead, hp) = initial_aead_and_hp(d_cid, Role::Client);
     let (_, pn) = remove_header_protection(&hp, header, payload);
 
-    let mut payload_enc = Encoder::with_capacity(MIN_INITIAL_PACKET_SIZE);
+    let mut write_buffer = Vec::with_capacity(MIN_INITIAL_PACKET_SIZE);
+    let mut payload_enc = Encoder::new(&mut write_buffer);
```

Comment:
`Encoder` no longer owns its `write_buffer`, thus allowing us to have `Encoder` encode into the long-lived `write_buffer` instead.

Downside, slightly more verbose test code.